File: go-ut-bench/datasets/go/go_code_files_self_contained/boundary/boundary_000.go
package main

func Clamp(value, minVal, maxVal int) int {
	if minVal > maxVal {
		panic("minVal must be less than or equal to maxVal")
	}
	if value < minVal {
		return minVal
	}
	if value > maxVal {
		return maxVal
	}
	return value
}

File: go-ut-bench/datasets/cpp/cpp_code_files_self_contained/boundary/boundary_000.cpp
#include <stdexcept>

int clamp(int value, int minVal, int maxVal) {
    if (minVal > maxVal) {
        throw std::invalid_argument("minVal must be less than or equal to maxVal");
    }
    if (value < minVal) {
        return minVal;
    }
    if (value > maxVal) {
        return maxVal;
    }
    return value;
}